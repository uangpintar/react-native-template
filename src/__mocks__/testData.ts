export const products = {
  products: [
    {
      id: 1,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
    {
      id: 2,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
    {
      id: 3,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
    {
      id: 4,
      title: 'test',
      description: 'test',
      thumbnail: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
      images: ['https://dummyjson.com/image/i/products/1/thumbnail.jpg'],
    },
  ],
};

export const users = {
  users: [
    {
      id: 1,
      firstName: 'first name',
      lastName: 'last name',
      image: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
    },
    {
      id: 2,
      firstName: 'first name',
      lastName: 'last name',
      image: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
    },
    {
      id: 3,
      firstName: 'first name',
      lastName: 'last name',
      image: 'https://dummyjson.com/image/i/products/1/thumbnail.jpg',
    },
  ],
};
