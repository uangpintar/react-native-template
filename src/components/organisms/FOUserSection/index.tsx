import React from 'react';
import { Box, ScrollView, Text } from 'native-base';
import { useGetUserListQuery } from '../../../services';
import { UsersCard } from '../../molecules';

export default function FOUserSection(): JSX.Element {
  const { isFetching, data, error } = useGetUserListQuery();

  return (
    <Box flex={1}>
      {error && <Text testID="error-text">There was an error</Text>}
      {isFetching && <Text>Loading</Text>}
      {data && (
        <ScrollView flex={1} p={'4'} testID={'scrollview'}>
          {data?.users?.map(
            (v: { lastName: string; firstName: string; image: string; id: number }) => (
              <UsersCard desc={v.lastName} name={v.firstName} image={v.image} key={v.id} />
            ),
          )}
        </ScrollView>
      )}
    </Box>
  );
}
