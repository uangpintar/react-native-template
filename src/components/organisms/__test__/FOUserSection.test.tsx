import { render } from '@testing-library/react-native';
import React from 'react';
import { wrapper } from '../../../__mocks__/wrapper';
import FOUserSection from '../FOUserSection';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
const server = setupServer(
  // Describe the requests to mock.
  rest.get('/users', (req, res, ctx) => {
    return res(ctx.json({}));
  }),
);
beforeAll(() => {
  // Establish requests interception layer before all tests.
  server.listen();
});
afterAll(() => {
  // Clean up after all tests are done, preventing this
  // interception layer from affecting irrelevant tests.
  server.close();
});

describe('FOUserSection Test', () => {
  test('should render FOUserSection correctly', async () => {
    const rendered = render(<FOUserSection />, { wrapper }).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
