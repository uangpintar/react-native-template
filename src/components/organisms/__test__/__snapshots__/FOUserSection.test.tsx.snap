// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FOUserSection Test should render FOUserSection correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Loading
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
